From 2428355c4716c18b599aa6a473065b3b09c96e16 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Wed, 10 Jul 2019 16:24:37 +0530
Subject: [PATCH 274/767] net: thunderx: Adjust CQE_RX drop levels for better
 performance

commit 6eb81fad89448434a377c216bacd415204b70113 from
git@git.assembla.com:cavium/WindRiver.linux.git

With the current RX RED/DROP levels of 192/184 for CQE_RX, when
packet incoming rate is high LLC is getting polluted resulting
in caches misses and higher latency in packet processing. This
slows down the whole process and performance loss. Hence reduced
reduced the levels to 224/216 (ie for a CQ size of 1024, Rx pkts
will be red dropped or dropped when unused CQE are less than 128
/160 respectively)

Change-Id: I1a19e4309644861e05400d78b625291d1dd53355
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/12340
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/12357
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/thunder/nicvf_queues.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/cavium/thunder/nicvf_queues.h b/drivers/net/ethernet/cavium/thunder/nicvf_queues.h
index bc2427c49b89..2460451fc48f 100644
--- a/drivers/net/ethernet/cavium/thunder/nicvf_queues.h
+++ b/drivers/net/ethernet/cavium/thunder/nicvf_queues.h
@@ -100,8 +100,8 @@
  * RED accepts pkt if unused CQE < 2304 & >= 2560
  * DROPs pkts if unused CQE < 2304
  */
-#define RQ_PASS_CQ_LVL         192ULL
-#define RQ_DROP_CQ_LVL         184ULL
+#define RQ_PASS_CQ_LVL         224ULL
+#define RQ_DROP_CQ_LVL         216ULL
 
 /* RED and Backpressure levels of RBDR for pkt reception
  * For RBDR, level is a measure of fullness i.e 0x0 means empty
-- 
2.31.1

